docs/development/AutomaticInstrumentation.md

```diff
@@ -235,3 +237,16 @@ Properties can be instrumented the same way as methods, but because of the way t
 string get_Name();
 void set_Name(string value);
 ```
+
+### Interfaces
+
+> ⚠️ Warning: This requires much more overhead because all of the types in a module must be inspected to determine if a module requires instrumentation. Carefully consider when to introduce this type of instrumentation.
```

Comment:
I don't have any numbers about overhead, but I don't think it is a problem that we need to jump in front of right away because we already have a good way to mitigate the overhead that is already in place:

When evaluating derived or interface instrumentations, we first check if the module we're inspecting has an assembly reference to the integration assembly. If not, then it's not possible for any types in the module to derive from the base class or implement the interface from the integration, so we skip processing.

In other words, the new RabbitMQ interface integration is only processed when the user references the `RabbitMQ.Client.dll`, so this adds no overhead to applications that do not use the `RabbitMQ.Client` NuGet package.

Reply:
And Tony brought up the line of code here that optimizes out this processing in the general case: https://github.com/DataDog/dd-trace-dotnet/pull/3256#discussion_r982751280